ui/hooks/useTransactionEventFragment.js

```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+
+import { useGasFeeContext } from '../contexts/gasFee';
+import { updateEventFragment } from '../store/actions';
+
+export const useTransactionEventFragment = () => {
+  const { transaction } = useGasFeeContext();
+
+  const updateTransactionEventFragment = useCallback(
+    (params) => {
+      if (!transaction) {
+        return;
+      }
+      updateEventFragment(`transaction-added-${transaction.id}`, params);
+    },
+    [transaction],
+  );
+
+  return {
+    updateTransactionEventFragment,
+  };
+};
```

Comment:
For transactions this hook provided only the option to update fragments, I found that other fragments changes were not required for transaction, not at least currently.